CONTRIBUTING.md

```diff
@@ -44,5 +44,35 @@ You are not required to run the system tests locally before creating a PR (they
 time and processing power), however if the tests fail on Travis you will need to update
 the PR until they pass.
 
+#### Adding a new library
+
 When you add a new library to build, make sure it is referenced in run-all.sh (for
-human contributors) and in .travis.yml (for continuous builds).
+human contributors) and in .travis.yml (for continuous builds). If your library has failures
+add the env row to the allow_failures section of .travis.yml and document the blockers.
+
+The structure for each library to test is as follows:
+
+```
+libraryBuilds/
+    - common/                   - Existing directory with common build config
+    - dependencyLib1/           - Existing root of a library that newLibrary depends on
+        - dependencyLib1/       - Existing project for a library that newLibrary depends on
+    - dependencyLib2/           - Ditto.
+        - dependencyLib2/       - Ditto.
+    - newLibrary/               - New directory for the root project
+        - newLibrary/           - New directory for the library itself
+            - build.gradle      - Build file containing j2objcTranslation directive for newLibrary
+        - dependencyLib1/       - Soft link to ../dependencyLib1/dependencyLib1/
+        - dependencyLib2/       - Soft link to ../dependencyLib2/dependencyLib2/
+        - settings.gradle       - 'include' directive for newLibrary and libraries it depends on
```

Comment:
Put settings.gradle last as that's the outlier while everything else is softlinks. That's also alphabetical ordering as well.